src/main/java/com/bootcamp/demo/service/UserService.java

```diff
@@ -0,0 +1,151 @@
+package com.bootcamp.demo.service;
+
+import com.bootcamp.demo.model.User;
+import com.bootcamp.demo.service.exception.ItemNotFoundException;
+import com.bootcamp.demo.service.exception.ServiceException;
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.QueryDocumentSnapshot;
+import com.google.cloud.firestore.WriteResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+@Service
+public class UserService implements IUserService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(UserService.class);
+    private static final String COLLECTION_USERS_PATH = "bookings/databases/users";
+    private final Firestore db;
+
+    public UserService(Firestore db) {
+        this.db = db;
+    }
+
+    @Override
+    public User createUser(final User user) {
+        LOGGER.info("CREATE USER - service function invoked: user = {}", user.toString());
```

Comment:
Similar to the above, see this thread for notes related to logging: https://github.com/nymphadora-tonks-team/booking-jb/pull/20#discussion_r762411991. Screenshot below as well in case it does not work ok. URLS:
- https://sematext.com/blog/java-logging-best-practices/
- https://www.geeksforgeeks.org/difference-between-hashig-and-encryption/

![image](https://user-images.githubusercontent.com/91753600/144911544-aa3b0e77-4990-4aec-8488-72f3c765a3b7.png)

Reply:
^ Do not mark the above as resolved if possible since I have referenced it in another PR & it might not redirect properly afterwards 🙂